### DISCUSSION TITLE
Combine rowSelection and Summary?

### DISCUSSION THREAD

**Discussion by ansyori28** [ant-design/ant-design#40351] [Q&A] [answered]
Hello, I have a Table with custom Summary placed in first row with sticky enabled, and the summary contains with input text to handle search. The table also need to implement rowSelection (type: checkbox) to select the rows. The problem is the first row (summary textbox) is started with position 0, so that the textbox count is not matches with the columns like in the screenshot below.

<img width="1667" alt="image" src="https://user-images.githubusercontent.com/8540792/213714667-a1946019-e78e-41bf-add9-e2bba847afd1.png">

How can I move the the summary row to start from position 1? so the search textbox is aligned with each columns. Here's the [CodeSandbox](https://codesandbox.io/s/antd-table-jf5ntb) for the details.

Any help will be much appreciated. Thanks!

**Comment by Anggieakbar025** [comment, marked as answer]
Try to set the first column of your summary as a blank column, then render the summary with input in the next column. I think it will work

**Reply by ansyori28** [reply]
<img width="1274" alt="image" src="https://user-images.githubusercontent.com/8540792/214017157-1f18a495-5cac-40b8-9b74-1b29e950084e.png">

I added a new `<Table.Summary.Cell index={0} key={0}></Table.Summary.Cell>` with index 0 before looping the column list as you suggested, it works! thanks a lot
